### ISSUE TITLE
NoneType is not callable with simple wrapped environment

### ISSUE THREAD

**Issue by drozzy** [astooke/rlpyt#159]
I am trying to wrap a simple dummy environment so that rlpyt will work on it (see code below).
But when I run this I get this error. What am I doing wrong here? Any tips appreciated!

P.S.: I hope I'm using the GymEnvWrapper correctly.

```
(rlpyt) andriy@silverslab  % python 1_bitflipping.py
Unable to import tensorboard SummaryWriter, proceeding without.
logger_context received log_dir outside of /Users/andriy/Projects/rlpyt/data: prepending by /Users/andriy/Projects/rlpyt/data/local/<yyyymmdd>/<hhmmss>/
2020-05-22 10:33:44.677567  | dummy_0 Runner  master CPU affinity: UNAVAILABLE MacOS.
2020-05-22 10:33:44.677723  | dummy_0 Runner  master Torch threads: 1.
using seed 684
Traceback (most recent call last):
  File "1_bitflipping.py", line 66, in <module>
    cuda_idx=args.cuda_idx
  File "1_bitflipping.py", line 30, in main
    runner.train()
  File "/Users/andriy/Projects/rlpyt/rlpyt/runners/minibatch_rl.py", line 301, in train
    n_itr = self.startup()
  File "/Users/andriy/Projects/rlpyt/rlpyt/runners/minibatch_rl.py", line 81, in startup
    world_size=world_size,
  File "/Users/andriy/Projects/rlpyt/rlpyt/samplers/serial/sampler.py", line 51, in initialize
    global_B=global_B, env_ranks=env_ranks)
  File "/Users/andriy/Projects/rlpyt/rlpyt/agents/dqn/dqn_agent.py", line 37, in initialize
    global_B=global_B, env_ranks=env_ranks)
  File "/Users/andriy/Projects/rlpyt/rlpyt/agents/base.py", line 84, in initialize
    **self.model_kwargs)
TypeError: 'NoneType' object is not callable
```

My code:

```python
from rlpyt.samplers.serial.sampler import SerialSampler
from rlpyt.agents.dqn.dqn_agent import DqnAgent
from rlpyt.algos.dqn.dqn import DQN
from rlpyt.envs.gym import GymEnvWrapper
from rlpyt.runners.minibatch_rl import MinibatchRlEval

def main(run_ID, cuda_idx):
    agent = DqnAgent()
    algo = DQN() 
    sampler = SerialSampler(
        EnvCls=GymEnvWrapper,
        env_kwargs={'env': DummyEnv()},
        batch_T=10, # Timesteps per sample batch
        batch_B=1,  # Num environments to run in parallel
        max_decorrelation_steps=0
    )

    runner = MinibatchRlEval(
        algo=algo,
        agent=agent,
        sampler=sampler,
        n_steps=1000
    )

    runner.train()
    
    from IPython import embed; embed(); exit(1)

import gym
class DummyEnv(gym.Env):
    """
    Runs env for 100 steps returning 0 reward except the last step returns 1
    """
    def __init__(self):
        self.n = 100
        self.action_space = gym.spaces.Discrete(2)
        self.observation_space = gym.spaces.Discrete(10)
        
    def reset(self):
        self.n = 100
        return 0

    def step(self, action):
        obs = 1
        self.n -= 1
        done = self.n <= 0
        rew = 1 if done else 0
        info = {}
        
        return obs, rew, done, info

if __name__ == '__main__':
    import argparse
    parser = argparse.ArgumentParser(formatter_class=argparse.ArgumentDefaultsHelpFormatter)
    parser.add_argument('--run_ID', help='run identifier (logging)', type=int, default=0)
    parser.add_argument('--cuda_idx', help='gpu to use', type=int, default=None)
    args = parser.parse_args()
    main(
        run_ID=args.run_ID,
        cuda_idx=args.cuda_idx
    )
```

**Comment by drozzy** [comment]
I hope I'm using the `GymEnvWrapper` correctly.

**Comment by astooke** [comment]
Almost!  But there needs to be a call to `gym.make()` inside there somewhere, see this line:

https://github.com/astooke/rlpyt/blob/85d4e018a919118c6e42fac3e897aa346d84b9c5/rlpyt/envs/gym.py#L163

or this example:  https://github.com/astooke/rlpyt/blob/85d4e018a919118c6e42fac3e897aa346d84b9c5/examples/example_2.py#L23

**Comment by drozzy** [comment]
Looking inside the [sampler](https://github.com/astooke/rlpyt/blob/36da682273777a6b5712e183a945d04ebeb22f4b/rlpyt/samplers/serial/sampler.py), where `EnvCls` is used, I don't see anything specific that requires gym. If I try to manually do what the sampler does, it works :)

<img width="697" alt="Screen Shot 2020-05-22 at 2 47 53 PM" src="https://user-images.githubusercontent.com/140710/82699757-3a19e300-9c3b-11ea-9cba-a55ee2b147f3.png">

**Comment by drozzy** [comment]
This is my own duplicate of another issue I created:
https://github.com/astooke/rlpyt/issues/164

That one has better examples and full source.
